src/Http/Http/src/WriteOnlyPipeStream.cs

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.IO.Pipelines
+{
+    /// <summary>
+    /// Represents a WriteOnlyStream backed by a PipeWriter
+    /// </summary>
+    public class WriteOnlyPipeStream : Stream
+    {
+        private PipeWriter _pipeWriter;
+
+        /// <summary>
+        /// Creates a new WriteOnlyStream
+        /// </summary>
+        /// <param name="pipeWriter"></param>
+        public WriteOnlyPipeStream(PipeWriter pipeWriter)
+        {
+            _pipeWriter = pipeWriter;
+        }
+
+        /// <inheritdoc />
+        public override bool CanSeek => false;
+
+        /// <inheritdoc />
+        public override bool CanRead => false;
+
+        /// <inheritdoc />
+        public override bool CanWrite => true;
+
+        /// <inheritdoc />
+        public override long Length => throw new NotSupportedException();
+
+        /// <inheritdoc />
+        public override long Position
+        {
+            get => throw new NotSupportedException();
+            set => throw new NotSupportedException();
+        }
+
+        /// <inheritdoc />
+        public override int ReadTimeout
+        {
+            get => throw new NotSupportedException();
+            set => throw new NotSupportedException();
+        }
+
+        /// <inheritdoc />
+        public override int Read(byte[] buffer, int offset, int count)
+            => throw new NotSupportedException();
+
+        /// <inheritdoc />
+        public override Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+          => throw new NotSupportedException();
+
+        /// <inheritdoc />
+        public override void Flush()
+        {
+            FlushAsync(default(CancellationToken)).GetAwaiter().GetResult();
+        }
+
+        /// <inheritdoc />
+        public override async Task FlushAsync(CancellationToken cancellationToken)
+        {
+            await _pipeWriter.FlushAsync(cancellationToken);
+        }
+
+        /// <inheritdoc />
+        public override long Seek(long offset, SeekOrigin origin)
+        {
+            throw new NotSupportedException();
+        }
+
+        /// <inheritdoc />
+        public override void SetLength(long value)
+        {
+            throw new NotSupportedException();
+        }
+
+        /// <inheritdoc />
+        public override void Write(byte[] buffer, int offset, int count)
+        {
+            WriteAsync(buffer, offset, count, default(CancellationToken)).GetAwaiter().GetResult();
```

Comment:
We should look at optionally blocking these sync APIs. @davidfowl

Reply:
Good catch!